be1-go/hub/base_channel.go

```diff
@@ -113,25 +118,41 @@ func (c *baseChannel) Catchup(catchup message.Catchup) []message.Message {
 		return messages[i].storedTime < messages[j].storedTime
 	})
 
-	result := make([]message.Message, 0, len(c.inbox.msgs))
+	result := make([]messageX.Message, 0, len(c.inbox.msgs))
 
 	// iterate and extract the messages[i].message field and
 	// append it to the result slice
 	for _, msgInfo := range messages {
-		result = append(result, *msgInfo.message)
+		result = append(result, msgInfo.message)
 	}
 
 	return result
 }
 
 // broadcastToAllClients is a helper message to broadcast a message to all
 // subscribers.
-func (c *baseChannel) broadcastToAllClients(msg message.Message) {
-	query := message.Query{
-		Broadcast: message.NewBroadcast(c.channelID, &msg),
+func (c *baseChannel) broadcastToAllClients(msg messageX.Message) {
+	// query := message.Query{
+	// 	Broadcast: message.NewBroadcast(c.channelID, msg),
+	// }
+
+	rpcMessage := message2.JSONRPC{
+		JSONRPC: "2.0",
+		Query: query.Query{
+			Method: "broadcast",
+			Broadcast: method.Broadcast{
+				Params: struct {
+					Channel string           `json:"channel"`
+					Message messageX.Message `json:"message"`
+				}{
+					Channel: c.channelID,
+					Message: msg,
+				},
+			},
+		},
 	}
 
-	buf, err := json.Marshal(query)
+	buf, err := json.Marshal(&rpcMessage)
```

Comment:
doesn't this result in an object looking like this? 
```
{
  "jsonrpc": "2.0",
  "Query": {
    ...
  }
}
```

Reply:
No, because it is embedded and not declared as a property.